llama-index-packs/llama-index-packs-code-hierarchy/examples/CodeHierarchyNodeParserUsage.ipynb

```diff
@@ -0,0 +1,1180 @@
+{
```

Comment:
<p>This is maybe a general design question, but what is the value of placing the node_id in the string when splitting? Wouldn't this just confuse the LLM when trying to do RAG over this?</p><p> </p><p> </p><p>I guess reading further down, this works best when you have an agent tool that allows for</p><p>a) general queries into the code</p><p>b) another tool that allows the agent to retrieve by ID</p><p> </p><p>Is this the correct takeaway?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/10671/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Reply:
You want the agent to be able to query either by code element names, or via the uuid placed in the comments by the skeletonizer. The same tool allows for both kinds of queries.